File: Python/Psd.py

from IArchivo import IArchivo

class Psd(IArchivo):

    def abrir(self, filename: str):

        # Aca estaria todo el algoritmo para descifrar el archivo en binario y poder abrirlo.
        print("Abri el archivo Psd de nombre '" + filename + "'")

File: C#/Navegador/Strategy/Psd.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Navegador.Strategy
{
    public class Psd : IArchivo
    {

        public void abrir(string filename)
        {

            // Aca estaria todo el algoritmo para descifrar el archivo en binario y poder abrirlo.
            Console.WriteLine("Abri el archivo Psd de nombre '" + filename + "'");

        }

    }
}
